### PR TITLE
upgrade pip, conda and setuptools in s2i image

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

It upgrades conda, pip and setuptools we use in python wrapper. 
This also fixes the 
```
  ModuleNotFoundError: No module named '_distutils_hack'
```
error reported in https://github.com/SeldonIO/seldon-core/issues/4209

**Which issue(s) this PR fixes**:

Fixes https://github.com/SeldonIO/seldon-core/issues/4209

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: wrappers/s2i/python/Makefile [hunk 1]
```diff
@@ -3,8 +3,8 @@ SHELL:=/bin/bash
 
 PYTHON_VERSION=3.8.10
 
-CONDA_DOWNLOAD_VERSION=4.7.12
-CONDA_VERSION=4.10.3
+CONDA_DOWNLOAD_VERSION=py38_4.12.0
```
Comment (RafalSkolasinski) [inline wrappers/s2i/python/Makefile:6]: For context, this is only about the python in downloaded conda, we still install desired version during the final image build.

**Comment by axsaucedo** [comment]
Nice one
/test integration

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 768e78c1fc3a0bc0744242fedefaa31669569484 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4210-integration-w4lsj) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 768e78c1fc3a0bc0744242fedefaa31669569484 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4210-integration-mkqk8) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Now failure only on `test_label_update[1.12.0]`

File: wrappers/s2i/python/Makefile (hunk 1 above)
Comment (axsaucedo) [reply wrappers/s2i/python/Makefile:6]: Would it be worth adding a comment with this?

**Comment by RafalSkolasinski** [comment]
Integration passed! So it was just flakiness.

**Comment by RafalSkolasinski** [comment]
image

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/4210#issuecomment-1183146154 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**